synapse/handlers/saml_handler.py

```diff
@@ -165,11 +170,13 @@ async def _map_saml_response_to_user(
                 saml2.BINDING_HTTP_POST,
                 outstanding=self._outstanding_requests_dict,
             )
+        except saml2.response.UnsolicitedResponse:
+            raise SynapseError(400, "Unexpected SAML2 login.")
```

Comment:
Will this make #7056 harder to solve? (Should we log the SAML session ID?)

Reply:
fair. Have added a log line.